google-cloud-firestore/src/main/java/com/google/cloud/firestore/FieldPath.java

```diff
@@ -80,7 +81,10 @@ static FieldPath fromDotSeparatedString(String field) {
     if (PROHIBITED_CHARACTERS.matcher(field).matches()) {
       throw new IllegalArgumentException("Use FieldPath.of() for field names containing '˜*/[]'.");
     }
-    return empty().append(field);
+    splitOnDots = true;
+    FieldPath returnFieldPath = empty().append(field);
+    splitOnDots = false;
+    return returnFieldPath;
   }
```

Comment:
when you override splitChildPath you don't need this change.

Reply:
There is two functionality associated here.
1)whenever we run following code then it internally call ```fromDotSeparatedString(String field)``` which is seprate
field by using period.
```
    Map<String, Object> deleteData = new HashMap<>();
    deleteData.put("user.abc.color", FieldValue.delete());
    ApiFuture<WriteResult> writeResultApiFuture = db.collection("users-test").document(docName).update(deleteData);
   
```

2)whenever we are using FieldPath.of("a.b", "c.d") so here sepration of field using period is not required if we seprate here using period then code does not working as expected.
```
FieldPath path = FieldPath.of("a.b", "c.d");
ApiFuture<WriteResult> writeResultApiFuture =
        db.collection("users-test").document(docName).update(path, FieldValue.delete());
```
so manage this thing we have to require splitOnDots flag.

Reply:
It's an error-prone approach to use a static property. If there are two instances of FieldPath classes they will intersect, which will lead to flaky errors. If you can't avoid using an extra flag, make it not static at least.

```
private boolean splitRequired = true;

@Override
public append() {
   super.append();
   splitRequired = false;
}
```